receiver/dotnetdiagnosticsreceiver/dotnet/event_header.go

```diff
@@ -0,0 +1,109 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dotnet
+
+import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dotnetdiagnosticsreceiver/network"
+
+type eventHeader struct {
+	metadataID        uint32
+	sequenceNumber    int32
+	captureThreadID   int64
+	captureProcNumber int32
+	threadID          int64
+	stackID           int32
+	payloadSize       int32
+	timestampDelta    int64
+}
+
+func parseEventHeader(r network.MultiReader, h *eventHeader) (err error) {
+	// EventPipeEventHeader.ReadFromFormatV4
+	var b byte
+	b, err = r.ReadByte()
+	if err != nil {
+		return
+	}
+	f := headerFlags(b)
+
+	if f.isSet(headerFlagMetadataID) {
+		h.metadataID, err = r.ReadCompressedUInt32()
```

Comment:
Currently the runtime will always use compression when writing, but it is legal according to the spec to write without compression. There aren't any plans to start doing that, but it could happen in the future. To be fully compliant with the spec a reader should check the flag of the block to see if it's compressed or not. It's up to you if you want to address that preemptively or wait and deal with it if or when it happens.

Reply:
Thank you, It's good to know that non-compression isn't used yet (I didn't see it used, but wasn't sure). Will make a note of this for later phases.